config/v1alpha1/types_backup.go

```diff
@@ -0,0 +1,131 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Backup provides configuration for performing backups of the openshift cluster.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type Backup struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec BackupSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status BackupStatus `json:"status"`
+}
+
+type BackupSpec struct {
+	// +kubebuilder:validation:Required
+	EtcdBackupSpec EtcdBackupSpec `json:"etcdBackupSpec,omitempty"`
+}
+
+type BackupStatus struct {
```

Comment:
do you want to type the status (eg successful run at X, backup located with filename X on PVC X) here too or just have it as a unstructured json?

Reply:
I initially had that in mind but it seems like there is a convention to not have `config.openshift.io` config types display or require a status block.

That’s also kind of why we have the `backup.openshift.io`  `EtcdBackup` type which tracks a single backup attempt with a status on whether each backup attempt is successful and which PVC it’s saving to. 

But if there’s anything else that’s more cumulative, perhaps we can add that to the `etcd.operator.openshift.io`'s type status.

Reply:
Agree. We largely don't have status on these.